src/components/Pagination.svelte

```diff
@@ -0,0 +1,124 @@
+<script context="module">
+  export const paginateData = (array, totalPages) => {
+    let result = [];
+    let localData = array.slice();
+    for (let i = totalPages; i > 0; i -= 1) {
+      result.push(localData.splice(0, Math.ceil(localData.length / i)));
+    }
+    return result;
+  };
+
+  export const makePages = (page, perPage, data) => {
+    if (data.length === 0) return [];
+    let total = data.length;
+    let currentPage = page;
+    let lastPage = Math.ceil(total / perPage);
+    let from = page + perPage * (page - 1);
+    let to = page * perPage;
+    let pages = paginateData(data, lastPage);
+    return {
+      total,
+      currentPage,
+      lastPage,
+      from,
+      to,
+      pages,
+    };
+  };
+
+  export const goToPage = (page, pages) => {
+    let from = 0;
+    for (let i = 0; i < page - 1; i += 1) {
+      from += pages[i].length;
+    }
+    let to = from + pages[page - 1].length;
+    return {
+      currentPage: page,
+      from,
+      to,
+      page: pages[page - 1],
+    };
+  };
+</script>
+
+<script>
+  import { createEventDispatcher } from "svelte";
+
+  export let currentPage;
+  export let lastPage;
+  export let from;
+  export let to;
+  export let total;
+  const dispatch = createEventDispatcher();
+
+  function range(size, startAt = 0) {
+    return [...Array(size).keys()].map((i) => i + startAt);
+  }
+
+  function changePage(page) {
+    if (page !== currentPage) {
+      dispatch("changePage", page);
+    }
+  }
+</script>
+
+<p>
+  Page
+  <code>{currentPage}</code>
+  of
+  <code>{lastPage}</code>
+  (<code>{from}</code>
+  -
+  <code>{to}</code>
+  on
+  <code>{total}</code>
+  items)
+</p>
+
+<div class="flex flex-col items-center my-12">
+  <div class="flex text-gray-700">
+    <div
```

Comment:
I think we should probably disable these in the case that user is already on the first/last page.

Though on second thought -- maybe it's best actually to just leave these out for now, as I don't think they add a lot of value over just showing the individual pages as links. We can consider re-adding them later.